nodetypes/tosca.nodes/WebApplication/NodeType.tosca

```diff
@@ -10,6 +10,7 @@ node_types:
     properties:
       context_root:
         type: string
```

Comment:
We should avoid changing the normative types. We create them according to the specification in the standard.

Reply:
I think this is a mistake in the TOSCA specification. The properties table is not consistent to the node type definition in [[1]](https://docs.oasis-open.org/tosca/TOSCA-Simple-Profile-YAML/v1.3/os/TOSCA-Simple-Profile-YAML-v1.3-os.html#DEFN_TYPE_NODES_WEBAPPLICATION), so I refer to the intepretation by the TOSCA parser of OpenStack in [[2]](https://github.com/openstack/tosca-parser/blob/master/toscaparser/elements/TOSCA_definition_1_0.yaml).

Reply:
Well, considering tool-specific interpretation is fine by me. But then I would encourage you to consider the interpretation by xOpera. In the end, they need to support the models at runtime and, furthermore, are compliant with TOSCA 1.3 (the OpenStack parser only 1.0 AFAIK).

Reply:
xOpera merely interpretates the node type definition, so xOpera if the property is required or not. What matters is the underlying Ansible playbooks, which is not currently not available. I checked the history of the TOSCA specification. This part doesn't change from TOSCA 1.0 to 1.3. The reason why I refer to the OpenStack parser is that it is widely used for a long time.

Reply:
xOpera merely interpretates the node type definition, so it doesn't care if the property is required or not. What matters is the underlying Ansible playbooks, which are currently not available. I checked the history of the TOSCA specification. This part doesn't change from TOSCA 1.0 to 1.3. The reason why I refer to the OpenStack parser is that it is widely used for a long time.